cu29_derive/src/lib.rs

```diff
@@ -142,84 +142,88 @@ pub fn copper_runtime(args: TokenStream, input: TokenStream) -> TokenStream {
                         }
-                    }
-                }
-                CuTaskType::Sink => {
-                    let input_culist_index = int2index(
-                        step.culist_input_index
-                            .expect("Sink task should have an input message index."),
-                    );
-                    quote! {
-                        {
-                            #comment_tokens
-                            let cumsg_input = &mut payload.#input_culist_index;
-                            #task_instance.process(&self.copper_runtime.clock, cumsg_input)?;
+                        CuTaskType::Sink => {
+                            let input_culist_index = int2index(
+                                step.culist_input_index
+                                    .expect("Sink task should have an input message index."),
+                            );
+                            quote! {
+                                {
+                                    #comment_tokens
+                                    let cumsg_input = &mut payload.#input_culist_index;
+                                    #task_instance.process(&self.copper_runtime.clock, cumsg_input)?;
+                                }
+                            }
                         }
-                    }
-                }
-                CuTaskType::Regular => {
-                    let input_culist_index = int2index(
-                        step.culist_input_index
-                            .expect("Sink task should have an input message index."),
-                    );
-                    let output_culist_index = int2index(
-                        step.culist_output_index
-                            .expect("Src task should have an output message index."),
-                    );
-                    quote! {
-                    {
-                        #comment_tokens
-                        let cumsg_input = &mut payload.#input_culist_index;
-                        let cumsg_output = &mut payload.#output_culist_index;
-                        cumsg_output.metadata.before_process = self.copper_runtime.clock.now().into();
-                        #task_instance.process(&self.copper_runtime.clock, cumsg_input, cumsg_output)?;
-                        cumsg_output.metadata.after_process = self.copper_runtime.clock.now().into();
-                    }
-                    }
-                }
-            };
+                        CuTaskType::Regular => {
+                            let input_culist_index = int2index(
+                                step.culist_input_index
+                                    .expect("Sink task should have an input message index."),
+                            );
+                            let output_culist_index = int2index(
+                                step.culist_output_index
+                                    .expect("Src task should have an output message index."),
```

Comment:
minor, but i noticed that these error messages refer to src/sink tasks and not Regular

```suggestion
                                    .expect("Regular task should have an input message index."),
                            );
                            let output_culist_index = int2index(
                                step.culist_output_index
                                    .expect("Regular task should have an output message index."),
```

Reply:
oh nicely spotted, fixing that, it would have been confusing for the user :)